docs/source/whatsnew/version5.rst

```diff
@@ -5,71 +5,69 @@
-since 2012, and pydb is not installable from PyPI.
-
-
-
-Replacement of readline in TerminalInteractiveShell and PDB
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+These features are provided by the Python library `prompt_toolkit
+<http://python-prompt-toolkit.readthedocs.io/en/stable/>`__, which replaces
+``readline`` throughout our terminal interface.
 
-IPython 5.0 now uses ``prompt_toolkit``. The
-``IPython.terminal.interactiveshell.TerminalInteractiveShell`` now uses
-``prompt_toolkit``. It is an almost complete rewrite, so many settings have
-thus changed or disappeared. The class keep the same name to avoid breaking
-user configuration for the options with names that are unchanged.
+Relying on this pure-Python, cross platform module also makes it simpler to
+install IPython. We have removed dependencies on ``pyreadline`` for Windows and
+``gnureadline`` for Mac.
 
-The usage of ``prompt_toolkit`` is accompanied by a complete removal of all
-code, using ``readline``. A particular effect of not using `readline` anymore
-is that `.inputrc` settings are note effective anymore. Options having similar
-effects have likely been replaced by a configuration option on IPython itself
-(e.g: vi input mode).
-
-The `PromptManager` class have been removed, and the prompt machinery simplified.
-See `TerminalInteractiveShell.prompts` configurable for how to setup your prompts.
-
-.. note::
-
-    During development and beta cycle, ``TerminalInteractiveShell`` was
-    temporarly moved to ``IPython.terminal.ptshell``.
-
-
-Most of the above remarks also affect `IPython.core.debugger.Pdb`, the `%debug`
-and `%pdb` magic which do not use readline anymore either.
+Backwards incompatible changes
+------------------------------
 
-The color handling has been slightly changed and is now exposed,
-in particular the colors of prompts and as you type
-highlighting can be affected by :
-``TerminalInteractiveShell.highlight_style``. With default
-configuration the ``--colors`` flag and ``%colors`` magic behavior
-should be mostly unchanged. See the `colors <termcolour>`_ section of
-our documentation
+- The ``%install_ext`` magic function, deprecated since 4.0, has now been deleted.
+  You can distribute and install extensions as packages on PyPI.
+- Callbacks registered while an event is being handled will now only be called
+  for subsequent events; previously they could be called for the current event.
+  Similarly, callbacks removed while handling an event *will* always get that
+  event. See :ghissue:`9447` and :ghpull:`9453`.
+- Integration with pydb has been removed since pydb development has been stopped
+  since 2012, and pydb is not installable from PyPI.
+- The ``autoedit_syntax`` option has apparently been broken for many years.
+  It has been removed.
+
+New terminal interface
+~~~~~~~~~~~~~~~~~~~~~~
+
+The overhaul of the terminal interface will probably cause a range of minor issues;
```

Comment:
Really like this section for the user 👍 

Suggested text, but fine as is too.

The terminal interface's overhaul (or update) may cause a range of minor issues for existing users. This is inevitable for such a significant change, and we've done our best to minimize these issues. Some issues that we are aware of as well as some suggestions for the user: